src/couch_replicator_scheduler.erl

```diff
@@ -451,10 +453,19 @@ reschedule(State) ->
     ok.
 
 
+-spec stop_excess_jobs(#state{}, non_neg_integer()) -> ok.
 stop_excess_jobs(State, Running) ->
     #state{max_jobs=MaxJobs} = State,
-    if Running > MaxJobs ->
-        stop_jobs(Running - MaxJobs, State);
+    StopCount = Running - MaxJobs,
+    if StopCount > 0 ->
+        Stopped = stop_jobs(StopCount, true, State),
+        OneshotLeft = StopCount - Stopped,
+        if OneshotLeft > 0 ->
+            stop_jobs(OneshotLeft, false, State),
```

Comment:
doesn't this violate the point of the patch? (that we won't stop one-shot replications).